mm2src/coins/utxo_signer/src/with_key_pair.rs

```diff
@@ -155,16 +155,17 @@ pub fn p2wpkh_spend(
 ) -> UtxoSignWithKeyPairResult<TransactionInput> {
     let unsigned_input = get_input(signer, input_index)?;
 
-    let script = Builder::build_p2pkh(&key_pair.public().address_hash().into());
-    if script != prev_script {
+    let script_code = Builder::build_p2pkh(&key_pair.public().address_hash().into()); // this is script code by BIP-0141
+    let witness_program = Builder::build_p2witness(&key_pair.public().address_hash().into());
+    if witness_program != prev_script {
```

Comment:
We need to pass the correct `prev_script` to `sign_tx` in multiple places otherwise this will cause some functionalities to not work correctly. e.g.
https://github.com/KomodoPlatform/komodo-defi-framework/blob/d1da866e2ae6284775ee527a92e50217721b353b/mm2src/coins/lightning/ln_events.rs#L222-L229
https://github.com/KomodoPlatform/komodo-defi-framework/blob/d1da866e2ae6284775ee527a92e50217721b353b/mm2src/coins/utxo.rs#L1806-L1812
And other places where `sign_tx` is used